src/bibgetter.py

```diff
@@ -341,17 +506,177 @@ def format(filename):
         stdout=subprocess.DEVNULL,
     )
 
+def substitute_bibtex_key(entry_text, expected_key):
+    """
+    Substitute the record identifier of a bibtex entry.
+
+    We use this to print a bibtex entry with the record identifier that was requested by the user,
+    regardless of the record identifier used in the central bibliography.
+    """
+    lines = entry_text.splitlines()
+    # Extract entry id from first line
+    entry_id = re.search(r'{\s*([^,\s]+)', lines[0]).group(1)
+    lines[0] = lines[0].replace('{' + entry_id, '{' + expected_key)
+    # Replace expected_key with entry_id in ids field
+    for i, line in enumerate(lines):
+        if re.match(r'^\s*ids\s*=', line):
```

Comment:
Will this only match lowercase `ids`? The content in the central bibliography isn't guaranteed to be sanitized like this.

Reply:
By default they are guaranteed to be (`<output_fieldcase>lower</output_fieldcase>` in biber-formatting.conf). Are the formatting options considered customizable by the user? Some biber formatting options like `--isbn-normalise` are embedded in the python code, so I mostly thought of biber-formatting.conf as part of the code, essentially.